sbi/inference/potentials/base_potential.py

```diff
@@ -83,36 +82,33 @@ def return_x_o(self) -> Optional[Tensor]:
         return self._x_o
 
 
-class CallablePotentialWrapper(BasePotential):
+class CustomPotential(Protocol):
+    """Protocol for custom potential functions."""
+
+    def __call__(self, theta: Tensor, x_o: Tensor) -> Tensor: ...
+
+
+class CustomPotentialWrapper(BasePotential):
     """If `potential_fn` is a callable it gets wrapped as this."""
 
     def __init__(
         self,
-        potential_fn: Callable,
+        potential_fn: CustomPotential,
         prior: Optional[Distribution],
         x_o: Optional[Tensor] = None,
         device: str = "cpu",
     ):
         """Wraps a callable potential function.
 
         Args:
-            potential_fn: Callable potential function, must have `theta` and `x_o` as
-                arguments.
+            potential_fn: Custom callable potential function.
```

Comment:
Maybe add some more details here:
- Function must have exactly two positional arguments where the first is theta and the second is the x_o

Reply:
Ahh, I see this is kinda through the protocol. But still providing minimal requirement to have 2 positional arguments might be nice.